src/rust/engine/process_execution/src/lib.rs

```diff
@@ -1076,13 +1081,58 @@ fn make_wrapper_for_append_only_caches(
                 "fi\n",
             ),
             quoted_path
-        )
-        .map_err(|err| format!("write! failed: {err}"))?;
+        ))
+    } else {
+        None
+    };
+
+    // Generate code to replace `{chroot}` markers if any are present in the command.
+    let sandbox_root_fragment = if let Some(token) = sandbox_root_token {
+        let fragment = format!(
+            concat!(
+                "sandbox_root=\"$(/bin/pwd)\"\n",
+                "args=(\"${{@//{0}/$sandbox_root}}\")\n",
+                "set -- \"${{args[@]}}\"\n",
+            ),
+            token
+        );
+        Some(fragment)
+    } else {
+        None
+    };
+
+    if caches_fragment.is_none() && sandbox_root_fragment.is_none() {
+        return Ok(None);
     }
 
-    // Finally, execute the process.
-    writeln!(&mut script, "exec \"$@\"").map_err(|err| format!("write! failed: {err:?}"))?;
-    Ok(script)
+    let script = {
+        let mut script = String::new();
+
+        let shebang = match sandbox_root_fragment {
+            Some(_) => "/bin/bash", // the array features need bash and not just plain old /bin/sh
+            _ => "/bin/sh",
```

Comment:
These paths (and others) ideally would be discovered in the remote execution environment, but the REAPI client is what runs those programs in the remote execution environment in the first place. Maybe we should make these configurable in the future?

Reply:
Or an alternate solution like download [BusyBox](https://www.busybox.net/) and use its `ash` shell.

Reply:
Hm, interesting approach. Definitely seems expedient.